### DISCUSSION TITLE
What's the best approach to internalisation of an app?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#9983] [Q&A] [answered]
Question as above. How to build and ship an Avalonia app with different language packs?

**Comment by timunie** [comment, marked as answer]
Hard to say what the BEST approach is. I personally like to use `DynamicResources` as I can easily swap out them easily. This may not be good for huge Apps though.  

Example: https://github.com/timunie/Tims.Avalonia.Samples#localization-sample

Other than that you can also use resx or resw files I think. More about it: Related: 
- https://github.com/AvaloniaUI/Avalonia/discussions/8741
- https://github.com/AvaloniaUI/Avalonia/discussions/9723
- https://github.com/AvaloniaUI/Avalonia/discussions/6974
- https://github.com/AvaloniaUI/Avalonia/discussions/7131

**Comment by lukaszossowski** [comment]
For my app I created a singleton AppViewModel which contains all the strings as properties which only have a getter with a switch inside, like this

```
  public string AppSettingsMessage
        {
            get
            {
                switch (language)
                {
                    case "pl": return "USTAWIENIA APLIKACJI";
                    case "de": return "de";
                    case "cz": return "cz";
                    case "fr": return "fr";
                    default: return "APP SETTINGS";
                }
            }
        }
```

The instance of this class is a property of my BaseViewModel, which all other VMs inherit from, so that they all have access to it. Then I can just bind to the strings like regular properties and have some other app-wide settings stored and managed there. It is a singleton so it doesn't get created with every new viewmodel (which is quite heavy on performance if there are a lot of properties in it) and to further mitigate any performance losses I am also using compiled bindings, though I am fairly sure that they are superfluous, I just like getting compile errors instead of looking for misspelled bindings.

**Comment by npolyak** [comment]
take a look at https://www.codeproject.com/Articles/5317972/Theming-and-Localization-Functionality-for-Multipl. Keep in mind that works only for 10.18. I plan to convert it to 11 once we have the next version of Avalonia 11 (hopefully 11 RC)
